raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -173,22 +178,34 @@ private void updateFollowerHighWatermark(FollowerState state, OptionalLong highW
             long newHighWatermark = Math.min(endOffset().offset, highWatermark);
             state.updateHighWatermark(OptionalLong.of(newHighWatermark));
             highWatermarkOpt.ifPresent(log::updateHighWatermark);
-            logger.trace("Follower high watermark updated to {}", newHighWatermark);
+            logger.debug("Follower high watermark updated to {}", newHighWatermark);
             applyCommittedRecordsToStateMachine();
         });
     }
 
-    private void updateLeaderEndOffset(LeaderState state) {
+    private void updateLeaderEndOffsetAndTimestamp(LeaderState state) {
         if (state.updateLocalEndOffset(log.endOffset())) {
-            logger.trace("Leader high watermark updated to {} after end offset updated to {}",
+            logger.debug("Leader high watermark updated to {} after end offset updated to {}",
                     state.highWatermark(), log.endOffset());
             applyCommittedRecordsToStateMachine();
         }
+
+        final OptionalLong lastFetchTimestamp = state.updateLocalFetchTimestamp(time.milliseconds());
```

Comment:
I feel we should consolidate with `maybeUpdateFetchTimerWithRemoteFetchTimestamp` by just pulling `quorum.localId` as a parameter to pass in.

Reply:
Ack.